ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        self._internal_coords = OrderedDict()
+
+    @property
+    def _all_coords(self):
+        """
+        Establish _all_coords as a property returning some _internal_coords.
+        """
+        return self._internal_coords
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self._internal_coords[name] = (physical_type, coords)
```

Comment:
I think this should raise an error if there is already a coordinate present of the same name.  Does the `OrderedDict` do that without us have to check that here?

But there should be a way to overwrite an existing coord.  That can of be done by first calling `remove` and then `add`.  but do you want a shortcut?  For example, a `replace=True/False` kwarg?

Reply:
Think `OrderedDict` guarantees that the order is preserved for the dict, so when iterating over we can rely on the same order every time. We can add an `update` as I have suggested at the very beginning instead of the `remove -> add` sequence as we are dealing with dicts.

Reply:
But @Cadair advised against adding an `'update'` function before... What do you guys think?

Reply:
Maybe we don't want the `update` functionality.  But we still need to know what happens when you try to `add` a coordinate with the same name as an existing coord.  As written, does it error?

Reply:
Sure, let me try now...

Reply:
The existing coord will get overwritten by the new argument with the same `name`... This is really neat!

Reply:
Added a test for this. 👍🏼

Reply:
Please read my initial comment more closely.  I don't think this is the behaviour we want.

Reply:
So now if we use the same name and update by doing `global_coords.add('name1', 'physical_type2', coord2)`, we will have the existing `global_coords` replaced (or overwritten)... Is this the expected behavior?

Reply:
This has been done by inheritance from some base properties.